src/stravalib/strava_model.py

```diff
@@ -6,7 +6,14 @@
 from datetime import datetime
 from typing import Literal, Optional
 
-from pydantic.v1 import BaseModel, Field
+from stravalib.pydantic_version import pydantic_major_version
```

Comment:
@soh55 this is the file that @jsamoocha was referring to that is autogenerated via the swagger API - [see here](https://github.com/stravalib/stravalib/pull/424) any changes made to this file will be overwritten until we update the actual model generation code workflow that jonatan designed and setup. (which works really well)

i think we all missed that in your previous PR because we were excited about a way to support both versions. 

i suspect youre keen on this change because we've pinned pydantic in our dependencies to avoid breaking changes in 2.0. we did that because it will take some time to upgrade the entire library to the new 2.0 API.

i hope that makes sense!